AI #2044: Update Editorial Notes section of Editorial Guidelines (#2111)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: guidelines/contributors/editorial-guidelines.md

Reviewer: @jpradocueva  
Please confer with @ijurica and determine what content from #2111 should be rolled into #1967.  I propose that pretty much everything having to do with normative requirements gets moved to #1967, and #2111 keeps a simple requirement that normative requirements are listed as bullets, and for all other details on normative requirements, to link to the normative requirements guidelines.

Reviewer: Discussed in Apr 6 Maintainers, and the group agreed to the above path forward.  We will delay Member approval a week to Apr 16 to allow us to approve #1967 on Apr 9 so that @jpradocueva can use that as a baseline here in #2111.

Reviewer: This is being resolved via #2191 which will be merged into this PR.

Reviewer: See [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1967/files#r3052946433).  What guideline shall we add for notes?  I'm not going to block on #1967, but we need a firm decision on this, which needs to be reflected in `AGENTS.md` -- otherwise, LLMs will keep tripping up.

> 3. Inconsistent Formatting of "Exceptions" and "Notes"
> Addressed with a different approach than the suggested blockquote standardization. The document now uses a consistent inline bold pattern:
> 
> **Exception for {Name}:** for named exceptions
> **Important Note:** for important callouts
> **Note:** / **Notes:** for informational callouts
>

Reviewer: Discussed with @jpradocueva.  He and @ijurica need to sync on the right rule(s) for this.

Reviewer: @jpradocueva @ijurica 
I did some digging on this.  MD036 only trips when you have a header without punctuation at the end.  So `**Notes:**` works but `**Notes**` doesn't.  [Here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/working_draft/specification/datasets/cost_and_usage/columns/serviceprovidername.md) is an example where we're using it in the spec already.

So we can use blockquotes for Notes and bold for Exceptions as proposed in TF-RM today.

Author: @shawnalpay I have revised the editorial guidelines for "Exceptions" based on your findings. Thanks!

        </td>
    </tr>
    <tr>

>         <td><strong>Normative Keywords</strong>(Formatting Only)</td>

Reviewer: ```suggestion
        <td><strong>Normative Keywords</strong> (Formatting Only)</td>
```

        <td>

>             Normative keywords and statements (see Normative Requirements Guidelines)

Reviewer: Lines 91 through 94 below are using `</br>`, but this is incorrect HTML; it needs to be `<br/>`.

* **Example** (Markdown, illustrative):

  ```md
  * PricingQuantity MUST be greater than 0.
  ```
* **No Compound Conditions:** Normative statements MUST NOT combine multiple conditions using “and” or “or” within a single statement.

---

### Entity References

* **Entity IDs:** Entity IDs (e.g., Column IDs, Attribute IDs, Dataset IDs) SHOULD be used in normative text sections, such as when specifying mandatory rules, schema definitions, or other implementation-related content. Entity IDs MUST be formatted without spaces and MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

* **Display Names:** Display Names SHOULD be used in introductory or explanatory sections where natural language context is more appropriate. These names should follow normal text conventions, including spaces between words (e.g., Commitment Discount ID).

* **Consistent Reference Type:** Normative statements MUST use Entity IDs consistently and MUST NOT mix Entity IDs and Display Names within the same statement.

* **Multiple Entity References:** When referencing multiple entities in a normative statement, all references MUST use Entity IDs.

> * **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).

Reviewer 2: I wonder if we should render this section in the same way we do Normative Requirements.

Reviewer: @davidearney 
While I agree with you completely, I wonder if there might be a larger refactor of the guidelines files ahead of us.  Some of them have requirements as bullets; otherwise do it the way it is here; and still others use no BCP14 statements of any kind.

We might be better served logging a Feature Request to capture this as a larger action and just leave it the way it is here for now.

@Matt-Cowsert What do you think?

Author: @davidearney, this list is the direct outcome of the normative requirements extraction process, capturing the issues we identified.

The current format was kept close to what we had before to maintain continuity, but I agree with your suggestion to break down some of the bullets to improve readability. Happy to incorporate those changes.

Given the limited time left in this cycle, the goal is to capture and stabilize the rules as they are. During the consistency review, we’ll cross-reference these with the Normative Requirements Guidelines.

Reviewer: @jpradocueva What AI or FR do we have to track that work?  If we don't have one, please record one before we resolve this thread.

Author: @shawnalpay, @ijurica asked me to create an issue to resolve this comment, AI #2141

> * **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).

Reviewer 2: Like this: 

```suggestion
* **Entity IDs:** Entity IDs (e.g., Column IDs, Attribute IDs, Dataset IDs) SHOULD be used in normative text sections, such as when specifying mandatory rules, schema definitions, or other implementation-related content. 

* **Entity IDs:** MUST be formatted without spaces.

* **Entity IDs:** MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

* **Display Names:** Display Names SHOULD be used in introductory or explanatory sections where natural language context is more appropriate. These names should follow normal text conventions, including spaces between words (e.g., Commitment Discount ID).

* **Consistent Reference Type:** Normative statements MUST use Entity IDs consistently.

* **Consistent Reference Type:** MUST NOT mix Entity IDs and Display Names within the same statement.

* **Multiple Entity References:** MUST use Entity IDs when referencing multiple entities in a normative statement.

* **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).
```

---

### Bullet Structure

* **Unordered List Style:** All unordered lists in Markdown MUST use asterisks (`*`) rather than dashes (`-`) or plus signs (`+`). This maintains visual consistency across the specification and aligns with our automated linting standards.

> * **Indentation Levels:** Nested bullet points MUST be indented using two spaces per level.

Reviewer 3: Worth having a note here about only stepping a level in after a composite (grouping) requirement ending in a ":"?

Author: @mike-finopsorg comment incorporated.

> * **Indentation Usage:** A nested indentation level MUST only be introduced when a parent bullet defines a composite (grouping) requirement, typically ending with a colon (:).

    </tr>
</table>

## Editorial Notes

> Note: Examples are illustrative, informative, and non-normative. They do not define requirements and do not override the rules in this section.

### Linking

* **Linking Only the First Time**: To prevent excessive linking within sections, Entity Names and Entity IDs (e.g., Column Names, Attribute IDs, and Glossary) will only be linked to their corresponding section or glossary the first time they appear in a section.

---

> ### Normative Statements

Reviewer: @ijurica's PR #1967 (and everywhere else, as I recall) refers to these as normative _requirements_, not _statements_.  Please change throughout.

Author: ```suggestion
### Normative Requirements
```

* **Single Requirement per Statement:** Each normative statement MUST express a single requirement.

* **Example** (Markdown, illustrative):

  ```md
  * ... MUST be of type String.
  * ... MUST conform to CurrencyFormat requirements.
  ```

* **Explicit Conditions:** Normative statements MUST include an explicit condition unless the rule applies in all cases.

* **Condition Format:** Conditional clauses MUST use standardized patterns: “when / if / unless / only when / only if / except when / except if <condition>”.

* **Example** (Markdown, illustrative):

  ```md
  * ... MUST be null when ChargeCategory is "Tax".
  * ... MUST be null if ChargeCategory is "Tax".
  * ... MUST be null unless ChargeCategory is "Usage".
  ```

* **Subject Consistency:** Each normative statement MUST clearly identify the subject being constrained.

* **Example** (Markdown, illustrative):

  ```md
  * PricingQuantity MUST be greater than 0.
  ```
* **No Compound Conditions:** Normative statements MUST NOT combine multiple conditions using “and” or “or” within a single statement.

---

### Entity References

> * **Entity IDs:** Entity IDs (e.g., Column IDs, Attribute IDs, Dataset IDs) SHOULD be used in normative text sections, such as when specifying mandatory rules, schema definitions, or other implementation-related content. 

Reviewer: I do not support prefixing all rules here with a subject in bold.  Propose stripping out all such occurences.

Reviewer: Discussed with @jpradocueva.  I will create a Feature Request to harmonize the formatting of all guidelines documents that @jpradocueva will own.  Will put larger format revisions across guidelines out of scope for 1.4 consistency review.  Resolving.

* **Entity IDs:** MUST be formatted without spaces.

* **Entity IDs:** MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

* **Display Names:** Display Names SHOULD be used in introductory or explanatory sections where natural language context is more appropriate. These names should follow normal text conventions, including spaces between words (e.g., Commitment Discount ID).

* **Consistent Reference Type:** Normative statements MUST use Entity IDs consistently.

> * **Consistent Reference Type:** MUST NOT mix Entity IDs and Display Names within the same statement.

Reviewer: Please scrub all normative statements like this that don't have an actor and being with the BCP14 statement.

Reviewer: Also, be careful using just the word "statement".  AI interpreted this line to mean "sentence" because it was missing the word "normative". 

Per another thread, you should adjust to use "normative requirements" instead of "normative statements" (or talk us out of it, and explain why).

Author: I will review the document and express it as normative requirements rather than normative statements

Author: All the references to "statements" have been removed or clarified.

* **Multiple Entity References:** MUST use Entity IDs when referencing multiple entities in a normative statement.

* **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).

> ---

Reviewer: We do not use these spacers in other guidelines files.  Please confer with @ijurica and determine if you will both use them or not use them in your PRs.

Author: ```suggestion

```

* **Non-Normative Use of “recommended”:** The lowercase word “recommended” may still be used in non-normative, descriptive text (e.g., “Feature level: Recommended”) provided it does not express a normative requirement on implementers.

* **Single Requirement per Statement:** Each normative statement MUST express a single requirement.

* **Example** (Markdown, illustrative):

  ```md
  * ... MUST be of type String.
  * ... MUST conform to CurrencyFormat requirements.
  ```

* **Explicit Conditions:** Normative requirements MUST include an explicit condition unless the rule applies in all cases.

* **Condition Format:** Conditional clauses MUST use standardized patterns: “when / if / unless / only when / only if / except when / except if <condition>”.

* **Example** (Markdown, illustrative):

  ```md
  * ... MUST be null when ChargeCategory is "Tax".
  * ... MUST be null if ChargeCategory is "Tax".
  * ... MUST be null unless ChargeCategory is "Usage".
  ```

* **Subject Consistency:** Each normative statement MUST clearly identify the subject being constrained.

* **Example** (Markdown, illustrative):

  ```md
  * PricingQuantity MUST be greater than 0.
  ```

> * **No Compound Conditions:** Normative requirements MUST NOT combine multiple conditions using “and” or “or” within a single statement.

Reviewer 4: This rule prohibits combining conditions with "and" or "or" within a single statement, but the PricingQuantity example at line 320 uses both:

> PricingQuantity MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and ChargeClass is not "Correction"

Suggest either scoping the rule (e.g., SHOULD avoid compound conditions where they reduce clarity) or splitting the example bullet to match.

Author: @Matt-Cowsert  This rule has been transferred to the Normative Requirements Guidelines. I keep this comment here to ensure the comment is addressed in the Normative Guidelines.

### Entity References

* **Entity IDs:** Entity IDs (e.g., Column IDs, Attribute IDs, Dataset IDs) SHOULD be used in normative text sections, such as when specifying mandatory rules, schema definitions, or other implementation-related content. 

* **Entity IDs:** MUST be formatted without spaces.

> * **Entity IDs:** MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

Reviewer: ```suggestion
* **Entity ID Spacing:** Entity IDs MUST be formatted without spaces.

* **Entity IDs Matching Schema:** Entity IDs MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).
```

* **Display Names:** Display Names SHOULD be used in introductory or explanatory sections where natural language context is more appropriate. These names should follow normal text conventions, including spaces between words (e.g., Commitment Discount ID).

> * **Consistent Reference Type:** Normative requirements MUST use Entity IDs consistently.

Reviewer 4: Two consecutive bullets share the bold prefix "Consistent Reference Type:" but describe different rules (use IDs consistently vs. don't mix IDs and Display Names). Distinct names would make them individually referenceable (e.g., "Consistent Reference Type" and "No Mixed Reference Types").

Author: @Matt-Cowsert, These rules have been moved out of the Editorial guidelines by PR #2191, which branched from this PR. These other rules are not incorporated into the Normative Requirements Guidelines. I will resolve the comment upon discussion with @ijurica

### Example

> **3.1.45. Pricing Quantity**
>
> The Pricing Quantity represents the volume of a given SKU associated with a resource or service used or purchased, based on the Pricing Unit. Distinct from Consumed Quantity (complementary to Consumed Unit), it focuses on pricing and cost, not resource and service consumption.
>
> **3.1.45.1. Requirements**
>
> PricingQuantity adheres to the following requirements:
> * PricingQuantity MUST be present in a Cost and Usage FOCUS dataset.
> * PricingQuantity MUST be of type Decimal.
> * PricingQuantity MUST conform to NumericFormat requirements.
> * PricingQuantity nullability is defined as follows:
>   * PricingQuantity MUST be null when SkuPriceId is null.
>   * PricingQuantity MUST be null when ChargeCategory is "Tax"
>   * PricingQuantity MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and ChargeClass is not "Correction"
>   * PricingQuantity MAY be null in all other cases.
> * PricingQuantity MUST be a valid decimal value when not null.

> > * Cost metric (e.g., ContractedCost) MUST equal the product of the corresponding unit price (e.g., ContractedUnitPrice) and PricingQuantity when the unit price is not null and

Reviewer 4: This requirement's condition continues into the next bullet ("PricingQuantity is not null."), which isn't a standalone normative statement. This conflicts with the "Single Requirement per Statement" rule at line 158. Suggest combining into one bullet with the full condition intact.

Author: This example was extracted from release v1.3. I have updated the example by aligning it with the new content in release v1.4.

* **Consistent Introductory Phrases:** Introductory phrases for normative lists SHOULD be consistent within a section.

### JSON Formatting

* **Valid JSON:** JSON examples MUST be valid and complete.

* **JSON Quotation Marks:** JSON keys MUST use double quotation marks.

* **Consistent Structure:** JSON examples SHOULD follow consistent formatting and indentation.

### Section Structure

* **Consistent Subsection Order:** Sections for the same entity type SHOULD follow the same subsection order (e.g., Title, Requirements, Entity ID, Display Name, Description, Content Constraints, Introduced (version)).

* **Consistent Subsection Titles:** Subsection titles SHOULD match exactly across sections of the same entity type.

* **Normative Scope:** Normative requirements (those using BCP-14 keywords) MUST appear only in the "Requirements" section.

* **No Normative Leakage:** Normative keywords (MUST, SHOULD, MAY, etc.) MUST NOT be used in any section other than "Requirements".

* **Example** (Markdown, illustrative):

```md
 ✔ Correct:
  ## Requirements
  ...
  * PricingQuantity MUST be greater than 0.
  
 ✘ Incorrect:
  ## Description
  ...
  * PricingQuantity MUST be greater than 0.

> ```
> 
> ### Examples
> 
> * **Valid Examples:** Examples MUST reflect valid combinations of values defined in the specification.

Author: ```suggestion
### Examples

> Note: Authors should consult the actual FOCUS attribute specification files as the source of truth, as these guidelines do not necessarily reflect the latest version.

* **Valid Examples:** Examples MUST reflect valid combinations of values defined in the specification.
```

    </tr>
</table>

## Editorial Notes

> Note: Examples are illustrative, informative, and non-normative. They do not define requirements and do not override the rules in this section.

### Linking

* **Linking Only the First Time:** To prevent excessive linking within sections, Entity Names and Entity IDs (e.g., Column Names, Attribute IDs, and Glossary) will only be linked to their corresponding section or glossary the first time they appear in a section.

### Entity References

* **Entity IDs Reference:** Formatting and usage conventions for normative requirements are defined in the Normative Requirements Guidelines.

* **Entity IDs Formatting:** Entity IDs MUST be formatted without spaces.

* **Entity IDs Naming:** Entity IDs MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

* **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).

### Column Values

> * **Inline Code for Values:** When referencing specific string values that a column can contain, the value MUST be enclosed in backticks to render as inline code (e.g., `Usage`). 

Reviewer: I heard enough concern from @kk09v in #2236 that I pulled the change from double-quotes to backticks from that PR.  We should strike it here and introduce it separately, after discussing in a future Maintainers meeting.

```suggestion
```

* **Display Names Formatting:** Display Names SHOULD follow normal text conventions, including spaces between words (e.g., Commitment Discount ID).

* **Consistent Formatting:** This formatting MUST be used consistently across all normative requirements, descriptions, and tables when referencing a specific string value.

### Comparison Terminology

To ensure consistent language when describing relationships and evaluations between values, normative requirements and standard prose MUST use the following terminology:

* **Numeric Values:** Use "equal" or "be equal to" when comparing numeric values, costs, quantities, or mathematical sums (e.g., "ContractedCost MUST be equal to BilledCost"). Do not use "match".
* **Identifiers and Strings:** Use "match" when comparing strings, IDs, or names (e.g., "HostProviderName MUST match ServiceProviderName"). Do not use "equal".
* **State and Conditions:** Use forms of the verb "to be" (e.g., "is", "are", "be") when evaluating if a column contains a specific value or state (e.g., "when ChargeCategory is `Purchase`" or "ChargeCategory MAY be `Usage`"). Do not use "equals" or "equal".

> * **Inequalities:** When expressing range or directional limits, the standard phrases "greater than or equal to" and "less than or equal to" MUST be used, even for non-numeric data types like timestamps or dates (e.g., "BillingPeriodLastUpdated MUST be greater than or equal to BillingPeriodCreated"). 

Reviewer: The "State and Conditions" rule says "Do not use 'equals' or 'equal'". However, the very next bullet ("Inequalities") explicitly requires the use of the phrase "greater than or equal to" for timestamps and dates. This will inevitably confuse contributors or cause automated linting tools to flag legitimate inequalities as violations.

We need to explicitly call out the inequalities phrasing as a valid exception within the "State and Conditions" bullet itself.

```suggestion
* **State and Conditions:** Use forms of the verb "to be" (e.g., "is", "are", "be") when evaluating if a column contains a specific value or state (e.g., "when ChargeCategory is `Purchase`" or "ChargeCategory MAY be `Usage`"). Do not use "equals" or "equal", **except** when evaluating inequalities (e.g., "greater than or equal to").
* **Inequalities:** When expressing range or directional limits, the standard phrases "greater than or equal to" and "less than or equal to" MUST be used, even for non-numeric data types like timestamps or dates (e.g., "BillingPeriodLastUpdated MUST be greater than or equal to BillingPeriodCreated"). 
```

* **Semantic Comparisons:** When comparing concepts, formats, or values that share the same meaning but may not be strictly identical strings, use "equivalent" or "semantically equivalent" (e.g., "PricingUnit MUST be semantically equivalent to the corresponding pricing measurement unit..."). Do not use "equal".

### Bullet Structure

* **Unordered List Style:** All unordered lists in Markdown MUST use asterisks (`*`) rather than dashes (`-`) or plus signs (`+`). This maintains visual consistency across the specification and aligns with our automated linting standards.

* **Indentation Levels:** Nested bullet points MUST be indented using two spaces per level.

* **Indentation Usage:** A nested indentation level MUST only be introduced when a parent bullet defines a composite (grouping) requirement, typically ending with a colon (:).

* **Example** (illustrative):

    ```md
    ... MUST adhere to the following rules:

    * ... MUST be of type JSON ...
    * ... MUST adhere to the following nullability requirements:
      * ... SHOULD NOT be null when ...
      * ... MUST be null when ...
      * ... MUST adhere to the following ...

>           * ... MUST be of type ...

Reviewer: Too many spaces.

```suggestion
        * ... MUST be of type ...
```

* **Inequalities:** When expressing range or directional limits, the standard phrases "greater than or equal to" and "less than or equal to" MUST be used, even for non-numeric data types like timestamps or dates (e.g., "BillingPeriodLastUpdated MUST be greater than or equal to BillingPeriodCreated"). 
* **Semantic Comparisons:** When comparing concepts, formats, or values that share the same meaning but may not be strictly identical strings, use "equivalent" or "semantically equivalent" (e.g., "PricingUnit MUST be semantically equivalent to the corresponding pricing measurement unit..."). Do not use "equal".

### Bullet Structure

* **Unordered List Style:** All unordered lists in Markdown MUST use asterisks (`*`) rather than dashes (`-`) or plus signs (`+`). This maintains visual consistency across the specification and aligns with our automated linting standards.

* **Indentation Levels:** Nested bullet points MUST be indented using two spaces per level.

* **Indentation Usage:** A nested indentation level MUST only be introduced when a parent bullet defines a composite (grouping) requirement, typically ending with a colon (:).

* **Example** (illustrative):

    ```md

>     ... MUST adhere to the following rules:

Reviewer: ```suggestion
    ... MUST adhere to the following requirements:
```

    </tr>
</table>

## Editorial Notes

> Note: Examples are illustrative, informative, and non-normative. They do not define requirements and do not override the rules in this section.

### Linking

* **Linking Only the First Time:** To prevent excessive linking within sections, Entity Names and Entity IDs (e.g., Column Names, Attribute IDs, and Glossary) will only be linked to their corresponding section or glossary the first time they appear in a section.

### Entity References

* **Entity ID References:** Formatting and usage conventions for normative requirements are defined in the Normative Requirements Guidelines.

* **Entity ID Formatting:** Entity IDs MUST be formatted without spaces.

* **Entity ID Naming:** Entity IDs MUST match the exact naming conventions used in the schema (e.g., CommitmentDiscountId).

* **Entity Display Name References:** References to FOCUS entities in non-normative content (i.e., outside Requirements sections) MUST use Display Names when available.

* **Entity Scope:** These rules apply to all entities defined in the FOCUS specification (e.g., Columns, Attributes, Datasets, Objects).

> ### Column Values

Reviewer: Is this the right name for this section?  Column Display Names don't have anything to do with column values.  Should this section perhaps to be called `Display Names` (or similar)?

> ### Column Values

Author: ```suggestion
### Display Names
```

  ```

### Notes and Exceptions

#### Notes

* **Blockquote Format:** Notes MUST be expressed using Markdown blockquote syntax (`>`). This creates a consistent, parseable container that can be styled during rendering.

* **Note Label:** Notes MUST begin with the label `Note:` or `Notes:` followed by a colon.

* **Single-Line Notes:** A note containing a single statement MUST be written as a single blockquote line.

* **Multi-Line Notes:** A note spanning multiple lines MUST repeat the blockquote prefix (`>`) on each line.

* **Multiple Notes:** When multiple notes are present in the same context, they SHOULD be grouped within a single blockquote and expressed as bullet points.

* **Bullet Usage in Notes:** When using bullets inside a note, the bullets MUST follow standard bullet formatting rules defined in this document.

* **Consistent Terminology:** Only the label `Note:` or `Notes:` MUST be used. Variants such as “Important Note”, “Warning”, or similar terms MUST NOT be used.

* **Non-Normative Content:** Notes MUST be informative and non-normative. They MUST NOT contain normative keywords (e.g., MUST, SHOULD, MAY).

> * **Rendering Flexibility:** Notes SHOULD be authored using blockquotes to allow downstream tooling to control visual presentation in HTML and PDF outputs.

Reviewer 5: ```suggestion
#### Notes

* **Permitted in Non-Normative Content Only:** Notes MUST NOT appear in Requirements sections. Notes MAY appear in non-normative sections of the specification and in guidelines documents.

* **Blockquote Format:** Notes MUST be expressed using Markdown blockquote syntax (`>`). This creates a consistent, parseable container and allows downstream tooling to control visual presentation in HTML and PDF outputs.

* **Note Label:** Notes MUST begin with the label `**Note:**` or `**Notes:**` (the term `Note` or `Notes`, followed by a colon, in bold).

* **Consistent Terminology:** Only the terms `Note` or `Notes` MUST be used. Variants such as `Important Note`, `Warning`, or similar terms MUST NOT be used.

* **Non-Normative Content:** Notes MUST be informative and non-normative. They MUST NOT contain normative keywords (e.g., MUST, SHOULD, MAY).

* **Single-Line Notes:** A note containing a single statement MUST be written as a single blockquote line.

* **Multi-Line Notes:** A note spanning multiple lines MUST repeat the blockquote prefix (`>`) on each line. A blank blockquote line (`>`) MUST separate the label from the bullet list.

* **Multiple Notes:** When multiple notes are present in the same context, they SHOULD be grouped within a single blockquote and expressed as bullet points.

* **Bullet Usage in Notes:** When using bullets inside a note, the bullets MUST follow standard bullet formatting rules defined in this document.
```

Reviewer: Works for me!

Author: ok, with the suggestions.

* **Example** (Markdown, illustrative):

  ```md
  > **Note:** This is a single-line note.
  ```

* **Example** (Markdown, illustrative):

  ```md
  > **Notes:**
  > * This is the first note.
  > * This is the second note, which provides additional information.

>   ```

Reviewer 5: **Linter compatibility check for Note formatting**

I tested formatting patterns by adding them to one of the column spec files, the following passed the linter check:

```markdown
> **Note:** This is a single-line note.
```

```markdown
> **Notes:**
>
> * This is the first note.
> * This is the second note, which provides additional information.
```

However, the following is problematic:

```markdown
> **Notes:**
> * This is the first note.
> * This is the second note, which provides additional information.
```

Fails with: `MD032/blanks-around-lists: Lists should be surrounded by blank lines`

*Note: I'm providing the updated markdown as a comment rather than as a direct suggestion commit, as nested code blocks don't render reliably inside GitHub suggestion blocks. Please copy and apply manually.*

```markdown
> **Notes:**
>
> * This is the first note.
> * This is the second note, which provides additional information.
```

Reviewer: +1.

Author: I have implemented the suggestion of adding an extra '>'

>   ```md
>   > **Notes:**

Author: ```suggestion
  ```md
  > **Notes:**
  >
```

* **No Blockquote Usage:** Exceptions MUST NOT be expressed using blockquote syntax.

* **Structural Relationship:** Exceptions MUST be directly associated with the requirement or content they modify and SHOULD appear immediately after the relevant content.

* **Formatting Approach:** Exceptions MUST be expressed using standard Markdown structures, such as plain text or bullet lists.

* **Bullet Usage:** When multiple exceptions are defined, they MUST be expressed as a bullet list.

* **Single Exception:** A single exception MAY be expressed as a standalone sentence or as a single bullet point.

> * **No Pseudo-Headers:** Exceptions MUST NOT use bold text or other formatting to simulate subsection headers (e.g., `**Exceptions:**`), as this violates linting rules.

Reviewer: Per my research, I was wrong about this.  As long as it has punctuation on the end, it works.  So `**Exceptions:**` works, but `**Exceptions**` does not.  Recommend we strike this line.

Reviewer: This was committed separately.  Resolving.

* **Subsection Usage:** Subsections (headers) MAY be used for exceptions only when necessary for clarity and when consistent with section structure rules.

* **Normative Scope:** Exceptions MUST NOT introduce or redefine normative requirements. If a condition represents a normative constraint, it MUST be expressed as a formal normative requirement instead.

> * **Avoidance in Normative Content:** Exceptions SHOULD be avoided in normative requirements. Authors SHOULD instead express conditions explicitly using structured requirement patterns (e.g., “when”, “unless”).

Reviewer 5: ```suggestion
#### Exceptions

* **Not Permitted in Requirements Sections:** Exceptions MUST NOT appear in Requirements sections of the specification. See [Normative Requirements Guidelines](normative-requirements-guidelines.md) for details.

* **Exception Format in Specification:** Exceptions in non-normative sections of the specification MUST be expressed using Markdown blockquote syntax (`>`).

* **Exception Format in Guidelines:** Exceptions in guidelines documents are not required to use blockquote syntax.

* **Exception Label:** Exceptions MUST begin with the label `**Exception:**` or `**Exceptions:**` (the term `Exception` or `Exceptions`, optionally followed by a qualifier, followed by a colon, in bold).

* **Consistent Terminology:** Only the term `Exception` or `Exceptions` MUST be used. Variants such as `Special Case`, `Caveat`, or similar terms MUST NOT be used.

* **Structural Relationship:** Exceptions MUST be directly associated with the content they modify and SHOULD appear immediately after the relevant content.

* **Single-Line Exceptions:** An exception containing a single statement MUST be written as a single line (or a single blockquote line in the specification).

* **Multi-Line Exceptions:** An exception spanning multiple statements MUST be expressed as a bullet list. When using blockquote syntax, a blank blockquote line (`>`) MUST separate the label from the bullet list.

* **Bullet Usage in Exceptions:** When using bullets inside an exception, the bullets MUST follow standard bullet formatting rules defined in this document.
```

Reviewer: I'm confused... I thought we agreed to using blockquotes for Notes and bold for Exceptions, but here you're saying Exceptions must be blockquoted?

Reviewer: Also, I would prefer to express Exceptions the same across the board.  Having a different rule for Exceptions in the spec vs guidelines will confuse me (and likely others).

Reviewer 5: Considering we've never used Exceptions in the specification outside Attributes Requirements sections (where they're no longer permitted anyway), do we even need to allow them in the specification at all?

If an exception truly needs to be flagged in a non-normative section of the specification, it could be expressed as a Note instead, right?

I suggest restricting Exceptions to guidelines only?

Reviewer: Gemini says:

> While the noun itself isn't in the Datasets or Columns sections, related variations of the word do appear there:
> * **"Except":** Used in the `ServiceName` requirements ("...except when no suitable ServiceCategory is available").
> * **"Exceptionally":** Used in the `BillingPeriodStatus` Implementation Context to describe the rare scenario of reopening a closed billing period ("Exceptionally, a previously closed billing period may be reopened...").

I'm open to this, but again, I'm concerned about changing course when we had agreement in TF-RM yesterday.  

I want to move this PR forward, but I'm concerned we don't have sufficient discussion and consensus on this particular point.  Could we perhaps strike the Exceptions language from this PR and put it forward in a separate PR for next week?

Reviewer 3: I agree with Shawn here, if we make an agreement in a synchronous meeting, then changing course from that agreement should be raised in slack as a discussion for asynchronous agreement prior to editing the PR, especially right before the PR is up for approval at members.

Reviewer 5: In TF-RM, we discussed two options for representing Exceptions (subheader vs. bold inline label), and both are captured in the PR for further consideration.

My understanding was that a final decision was not made, as we were not fully certain what the linter allows regarding inline bold formatting, and that I would bring forward a proposal after reviewing the options and validating linter constraints.

In this suggestion, I proposed an approach that also addresses a few additional issues I identified, and reflects my preference for formatting.

Reasoning:
- Using subheaders in guidelines did not seem practical, as we cannot have multiple "Exceptions" subheaders within a single file.
- I confirmed that the linter has no issues with a bold inline label followed by a colon.
- Aligning the formatting of Exceptions and Notes in non-normative content felt more consistent than aligning formatting between the spec and the guidelines.

As mentioned, I am open to suggestions.

Reviewer: Here is what I propose.  It's mostly what you proposed, except not using blockquotes

1. **Drops blockquotes for Exceptions entirely,** reflecting what I understood to be our agreement coming out of TF-RM.  (Clearly we don't have a common understanding of the outcome of yesterday's meeting -- but that's what I said, and heard.)
2. **Bans the `#### Exceptions` header,** which solves the duplicate-header linter errors that Irena encountered.
3. **Unifies the rules,** ensuring we don't have a "split-brain" standard where the spec and the guidelines follow different formatting rules.
4. **Bans Exceptions from Requirements sections,** upgrading the weak "SHOULD be avoided" to a hard "MUST NOT". This forces authors to use proper BCP-14 conditional logic (e.g., `when`, `unless`) instead of burying constraints in exceptions.

I'm more than happy to discuss this further in TF-RM next week -- we don't _have_ to approve and merge this PR tomorrow.  But if we can't find consensus on this before the Members meeting tomorrow, I would prefer simply to strike Exceptions guidelines completely and then issue a separate PR with a path forward, so we can start leveraging all the other good work we have this PR for the rest of consistency review.

```suggestion
#### Exceptions

* **Not Permitted in Requirements Sections:** Exceptions MUST NOT appear in Requirements sections of the specification. If a condition represents a normative constraint, authors MUST express it explicitly using structured requirement patterns (e.g., “when”, “unless”) instead.

* **No Blockquote Usage:** Exceptions MUST NOT be expressed using blockquote syntax (which is strictly reserved for Notes).

* **No Header Usage:** Exceptions MUST NOT use Markdown headers (e.g., `#### Exceptions`). 

* **Exception Label:** Exceptions MUST begin with the inline label `**Exception:**` or `**Exceptions:**` (the term `Exception` or `Exceptions`, optionally followed by a qualifier, followed by a colon, in bold).

* **Consistent Terminology:** Only the term `Exception` or `Exceptions` MUST be used. Variants such as `Special Case`, `Caveat`, or similar terms MUST NOT be used.

* **Structural Relationship:** Exceptions MUST be directly associated with the content they modify and SHOULD appear immediately after the relevant content.

* **Single-Line Exceptions:** An exception containing a single statement MUST be written as a single line of plain text immediately following the label.

* **Multi-Line Exceptions:** An exception spanning multiple statements MUST be expressed as a bullet list. A blank line MUST separate the label from the bullet list.

* **Bullet Usage in Exceptions:** When using bullets inside an exception, the bullets MUST follow standard bullet formatting rules defined in this document.
```

Reviewer 5: Overall I'm on board with this approach.

That said, I believe the `immediately following the label` in case of multi-line exceptions will trigger `MD032/blanks-around-lists`, as lists need to be surrounded by blank lines.

Suggestion:

```markdown
* **Multi-Line Exceptions:** An exception spanning multiple statements MUST be expressed as a bullet list. A blank line MUST separate the label from the bullet list.
```

Reviewer: @ijurica Good call.  Edited my suggestion above to incorporate that.

Author: I am ok with these changes.

* **Example** (Markdown, illustrative):

  ```md

>   Exceptions:

Reviewer: ```suggestion
  **Exceptions:**
```

Reviewer: Was committed separately.  Resolving.

  * This rule does not apply when ChargeCategory is `Adjustment`.
  * This rule does not apply when ChargeCategory is `Tax`.

>   ```

Reviewer 5: **Linter compatibility check for Exception formatting**

I tested the following formatting patterns by adding them to one of the column spec files, and they passed the linter check:

```markdown
**Exception:** This is a single-line exception.
```

```markdown
**Exceptions:**

* This is the first exception.
* This is the second exception.
```

Reviewer: Yes.  As I mentioned in a couple other threads: `**Exceptions:**` works but `**Exceptions**` does not because MD036 is okay with headers like this as long as they have punctuation at the end.

Reviewer 5: Note: I'm providing this as a comment rather than as a direct suggestion commit, as nested code blocks don't render reliably inside GitHub suggestion blocks. @jpradocueva if you agree with the suggestion, please copy and apply manually.

---

* **Example** (Markdown, illustrative — specification, single-line):

```markdown
> **Exception:** This is a single-line exception.
```

* **Example** (Markdown, illustrative — specification, multi-line):

```markdown
> **Exceptions:**
>
> * This is the first exception.
> * This is the second exception.
```

* **Example** (Markdown, illustrative — guidelines, single-line):

```markdown
**Exception:** This is a single-line exception.
```

* **Example** (Markdown, illustrative — guidelines, multi-line):

```markdown
**Exceptions:**

* This is the first exception.
* This is the first exception.
```

Author: I have applied the two examples without using blockquotes, as the group has not agreed on them. Further discussion is required.

* **Indentation Levels:** Nested bullet points MUST be indented using two spaces per level.

* **Indentation Usage:** A nested indentation level MUST only be introduced when a parent bullet defines a composite (grouping) requirement, typically ending with a colon (:).

* **Example** (illustrative):

    ```md
    ... MUST adhere to the following requirements:

    * ... MUST be of type JSON ...
    * ... MUST adhere to the following nullability requirements:
      * ... SHOULD NOT be null when ...
      * ... MUST be null when ...
      * ... MUST adhere to the following ...
        * ... MUST be of type ...
    ```

* **No Skipped Levels:** Nested bullets points MUST NOT skip indentation levels.

* **Consistent Indentation:** All bullet points within the same list MUST use consistent indentation.

* **Example** (Markdown, illustrative):

  ```md
  * ... MUST be null when ChargeCategory is "Tax".
  * ... MUST be null when ChargeCategory is "Adjustment".
  ```

### Formatting

> * **Introductory Phrase for Rules:** Normative bullet lists SHOULD be preceded by a short introductory phrase ending with a colon (:).

Reviewer: ```suggestion
* **Introductory Phrase for Requirements:** Normative bullet lists SHOULD be preceded by a short introductory phrase ending with a colon (:).
```

Reviewer 5: @jpradocueva @shawnalpay please consider this

```suggestion
* * **Structural Anchor Requirements:** Normative bullet lists MUST be preceded by a structural anchor requirement. See [Normative Requirements Guidelines](normative-requirements-guidelines.md#structural-anchor-requirement) for details..
```

Reviewer: I'm fine with this, but @jpradocueva is keen on starting all requirements in the guidelines with a subject (which I am not onboard with, but I've agreed to postpone my opposition on this until he tackles the harmonization of all guidelines files in the next release per FR #2237).

Author: This change it is ok with me.

* **Example** (Markdown, illustrative):

  ```md

>   ... MUST adhere to the following rules:

Reviewer: ```suggestion
  ... MUST adhere to the following requirements:
```

Reviewer 5: Alternative:

@jpradocueva, If you accept the suggestion above, which points to Normative Requirements Guidelines for details, please also remove the **Example** block and the **Consistent Introductory Phrases** bullet below.
